ROADMAP.md

```diff
@@ -0,0 +1,185 @@
+https://github.com/libp2p/rust-libp2p/pull/2289.
+
+Long term there is lots more to do, see tracking issue
+https://github.com/libp2p/rust-libp2p/issues/2883. Though this will be tackled as a new roadmap
+item.
+
+## WebRTC support (browser-to-server)
+
+| Status      | Target Completion | Tracking                                 | Dependencies                                   | Dependents |
+|-------------|-------------------|------------------------------------------|------------------------------------------------|------------|
+| In progress | Q4/2022           | https://github.com/libp2p/specs/pull/412 | https://github.com/libp2p/test-plans/issues/53 |            |
+
+
+We are currently implementing WebRTC for **browser-to-server** connectivity in
+https://github.com/libp2p/rust-libp2p/pull/2622. More specifically the server side. This will enable
+browser nodes to connect to rust-libp2p nodes where the latter only have self-signed TLS
+certificates. See https://github.com/libp2p/specs/pull/412 for in-depth motivation.
+
+Long term we should enable rust-libp2p running in the browser via WASM to use the browser's WebRTC
+stack. Though that should only happen after improved WASM support, see below.
+
+## Cross Behaviour communication
+
+| Status | Target Completion | Tracking                                          | Dependencies                                      | Dependents                                    |
+|--------|-------------------|---------------------------------------------------|---------------------------------------------------|-----------------------------------------------|
+| todo   | Q1/2023           | https://github.com/libp2p/rust-libp2p/issues/2680 | https://github.com/libp2p/rust-libp2p/issues/2832 | [Kademlia client mode](#kademlia-client-mode) |
+
+Today `NetworkBehaviour` implementations like Kademlia, GossipSub or Circuit Relay v2 can not
+communicate with each other, i.e. can not make use of information known by another
+`NetworkBehaviour` implementation. First step would be to exchange standard information about remote
+peers (e.g. supported protocols).
+
+Long term we might consider a generic approach for `NetworkBehaviours` to exchange data. Though that
+would deserve its own roadmap item.
+
+## Generic connection management
+
+| Status | Target Completion | Tracking                                          | Dependencies | Dependents |
+|--------|-------------------|---------------------------------------------------|--------------|------------|
+| todo   | Q1/2023           | https://github.com/libp2p/rust-libp2p/issues/2824 |              |            |
+
+See https://github.com/libp2p/rust-libp2p/issues/2824 for motivation. Given that this will enable
+downstream users to easier integrate with rust-libp2p, I think this counts as a "improving existing
+components" over "introducing a new component".
+
+First draft is in https://github.com/libp2p/rust-libp2p/pull/2828
```

Comment:
My thoughts on this one are similar to cross-behaviour communication but it is a little less clear because you need to work around the fact that connections get established first and a behaviour would have to instantly kill it again. This might get the job done but is a bit hacky.

I am inclined to say that this should either also be removed or grouped under a "Developer ergonomics" heading.

Reply:
> This might get the job done but is a bit hacky.

As an aside, that is how https://github.com/sigp/lighthouse and https://github.com/paritytech/substrate do it today.

> I am inclined to say that this should either also be removed or grouped under a "Developer ergonomics" heading.

In my eyes, this goes beyond ergonomics. I.e. it enables users to implement a clean eclipse defense.

Reply:
Perhaps it should go under a "Security" heading then?